niceplots/utils.py

```diff
@@ -1,127 +1,120 @@
-        raise TypeError(f"set_dark_background value of {set_dark_background} is invalid")
-
-
-def get_niceColors():
-    # Define an ordered dictionary of some nice Doumont style colors to use as the default color cycle
-    niceColors = OrderedDict()
-    niceColors["Yellow"] = "#e29400ff"  # '#f8a30dff'
-    niceColors["Blue"] = "#1E90FF"
-    niceColors["Red"] = "#E21A1A"
-    niceColors["Green"] = "#00a650ff"
-    niceColors["Maroon"] = "#800000ff"
-    niceColors["Orange"] = "#ff8f00"
-    niceColors["Purple"] = "#800080ff"
-    niceColors["Cyan"] = "#00A6D6"
-    niceColors["Black"] = "#000000ff"
-    # The 2 colours below are not used in the colour cycle as they are too close to the other colours.
-    # Grey is kept in the dictionary as it is used as the default axis/tick colour.
-    # RedOrange is the old Orange, and is kept because I think it looks nice.
-    niceColors["Grey"] = "#5a5758ff"
-    niceColors["RedOrange"] = "#E21A1A"
-
-    return niceColors
-
-
-def get_delftColors():
-    # Define an ordered dictionary of the official TU Delft colors to use as the default color cycle
-    delftColors = OrderedDict()
-    delftColors["Cyan"] = "#00A6D6"  # '#f8a30dff'
-    delftColors["Yellow"] = "#E1C400"
-    delftColors["Purple"] = "#6D177F"
-    delftColors["Red"] = "#E21A1A"
-    delftColors["Green"] = "#A5CA1A"
-    delftColors["Blue"] = "#1D1C73"
-    delftColors["Orange"] = "#E64616"
-    delftColors["Grey"] = "#5a5758ff"
-    delftColors["Black"] = "#000000ff"
-
-    return delftColors
+    curDir = os.path.dirname(os.path.abspath(__file__))
+    styleFile = os.path.join(curDir, "styles", styleName + ".mplstyle")
+
+    if afterReset:
+        rcdefaults()
+
+    # Set the style
+    style.use(styleFile)
+
+    # Get the colors and their names from the rcParams
+    return get_colors()
+
+
+@contextlib.contextmanager
+def styleContext(styleName="doumont-light", afterReset=False):
+    """
+    Temporarily change the style of plots. This function is a context manager,
+    so is to be used in a with block. For example::
+
+        with niceplots.styleContext("james-dark"):
+            # plot stuff here and it will be in the james style
+
+    Parameters
+    ----------
+    styleName : str, optional
+        Name of desired style. Avaiable styles are:
+
+            - doumont-light: the niceplots style you know and love
+            - doumont-dark: the dark version of the niceplots style you know and love
+            - james-dark: a really cool alternative to classic niceplots
+            - james-light: a version of james with a light background, naturally
+
+    afterReset : bool, optional
+        If True, will first reset the rcParams to the matplotlib then apply
+        the new rcParams. The alternative (when afterReset is False) is to apply
+        the new rcParams on top of the current rcParams.
+    """
+    with rc_context():
+        _ = setStyle(styleName, afterReset)
+        yield
+
+
+def get_colors():
```

Comment:
I know it pains you but can we make this `getColors` to be consistent with the rest of the functions

Reply:
I was consider this but unsure since the rest of the functions are so inconsistent. The previous get colors functions had underscores as do the rest of the functions in utils. I'm happy to do either, but I'm not sure which is _actually_ more "consistent".